lib/std/reset_event.zig

```diff
@@ -431,11 +431,33 @@ test "std.ResetEvent" {
             self.in.wait();
             assert(self.value == 3);
         }
+
+        fn sleeper(self: *Self) void {
+            self.in.set();
+            time.sleep(time.ns_per_ms);
+            self.value = 5;
+            self.out.set();
+        }
+
+        fn timedWaiter(self: *Self) !void {
+            self.in.wait();
+            testing.expectError(error.TimedOut, self.out.timedWait(time.ns_per_us));
+            try self.out.timedWait(time.ns_per_ms * 2);
+            testing.expect(self.value == 5);
+        }
     };
 
     var context = Context.init();
     defer context.deinit();
     const receiver = try std.Thread.spawn(&context, Context.receiver);
     defer receiver.wait();
     context.sender();
+
+    if (builtin.link_libc) {
```

Comment:
Why?

Reply:
@LemonBoy [It fails](https://dev.azure.com/ziglang/zig/_build/results?buildId=10152&view=logs&j=3e8797c7-5b0a-5f2c-07a4-1bc5e60a122e&t=c222f6c4-0121-56a3-9785-67eccb26a308&l=19891), I did not investigate it deeply since the change I made is not even applicable on that platform. I guess its possible there is another latent issue there, but I do not even have a development environment for Windows. It is also possible that the issue was just two short of a time for the first thread to sleep, I originally had it at only 10,000 ns. This kind of test can be a nightmare for build reliability so I bumped the wait up to a whole ms and rebased.

Reply:
> It fails

That's not a good reason to mask the test flakiness.

> It is also possible that the issue was just too short of a time for the first thread to sleep

Keep in mind that on Windows `sleep()` has a resolution of `ms` rather than `ns`, everything less than that is sliently flushed to zero, and the [kernel](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-sleep) interprets that as a request to give away your time slice to some other thread (is this a bug? a feature?).

The sleep amount depends on so many factors such as the HW being used and the actual load (and the task switching frequency and ...), I got the test to reliably pass (out of 100 tries) by having `sleeper` sleep for 2ms and `timedWaiter` wait for 5ms.

Reply:
@LemonBoy Thanks for trying that, we definitely need much longer times than I started with, and the libc thing was indeed a red herring. I bumped the `timedWaiter` out to 10ms after we had a failure on one platform with 2/5. We shouldn't usually wait anywhere near that long, and I suppose it could be made even longer considering anytime we exceed that the entire build will fail. Now the Linux build [is failing](https://dev.azure.com/ziglang/zig/_build/results?buildId=10183&view=logs&j=9512b82f-d185-50ea-ee23-d010bc14782f&t=ca25486c-bdcc-50c6-de82-a9ccd6b64318&l=1067) with a totally new error now ` expected LLVM 11.x but found 11.0.1 using /usr/bin/llvm-config-11`

Reply:
> Thanks for trying that, we definitely need much longer times than I started with, and the libc thing was indeed a red herring.

No problem, getting this timing-sensitive tests to work reliably on the CI machines requires a bit of fiddling.
The generous wait amount should be enough to let the thread complete even on "overcrowded" machines.

> Now the Linux build is failing with a totally new error now expected LLVM 11.x but found 11.0.1 using /usr/bin/llvm-config-11

That's unrelated to this change, something in the build system went wrong and is making all the builds fail...